tests/test_visitors/test_ast/test_classes/test_wrong_empty_lines_count.py

```diff
@@ -0,0 +1,108 @@
+"""Test method contain only allowed empty lines count."""
+import pytest
+
+from wemake_python_styleguide.violations.best_practices import (
+    WrongEmptyLinesCountViolation,
+)
+from wemake_python_styleguide.visitors.ast.function_empty_lines import (
+    WrongEmptyLinesCountVisitor,
+)
+
+class_with_wrong_method = """
+class WrongClass(object):
+
+    def wrong_method(self):
+        foo()
+
+        bar()
+
+        baz()
+
+        lighter()
+"""
+
+class_with_allow_method = """
+class WrongClass(object):
+
+    def wrong_method(self):
+        foo()
+        bar()
+        baz()
+"""
+
+wrong_function = """
+def func():
+    foo()
+
+    a = 1 + 4
+
+    baz()
+"""
+
+wrong_function_with_loop = """
```

Comment:
What about edge cases like below ones?

```python
def foo():
    return (
        pekaboo['attr'] == 'dummy-value'
        and bar() == 'Knights that say Ni!'


        or (
            'attr2' in pekaboo
            and not some_object.attribute.method()
        )
    )
```

```python
pekaboo(
    foo=123,



    bar='abc',
)
```

Can they be handled by the `ast`?
Probably this violation should be implemented by using `BaseTokenVisitor`

Reply:
Yeh, I plan to change visitor to `BaseTokenVisitor` because `ast` ignore comments. I check these cases after changing the visitor, thank you